### PR TITLE
Skip broken values during storage iteration

### LINKED ISSUE #2200: Skip broken values during storage iteration
### Issue To Be Solved

As described in https://github.com/onflow/cadence/issues/2170, some storage values could be broken/fail to load, due to outdated contracts. This becomes problematic in cases like storage iteration etc.

### Suggested Solution

Improve the storage iteration API to skip broken values

### PR DESCRIPTION
**Pull request by SupunS**
Closes #2200

## Description

The storage iteration API will now skip any broken values in the storage.

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/interpreter/interpreter.go [hunk 1]
```diff
@@ -3271,8 +3276,31 @@ func (interpreter *Interpreter) newStorageIterationFunction(addressValue Address
...
+							switch r := r.(type) {
+							case errors.UserError, errors.ExternalError:
+								typeError = r.(error)
+							default:
+								panic(r)
+							}
+						}
+					}()
+					_, typeError = inter.ConvertStaticToSemaType(staticType)
+				}()
```
Comment (dsainati1) [inline runtime/interpreter/interpreter.go:3296]: is it necessary to have this double nested function here? What errors are we throwing inside `ConvertStaticToSemaType` that aren't being returned in the `err` value?

Comment (SupunS) [reply runtime/interpreter/interpreter.go:3296]: Yup, the nested function is needed to recover errors thrown from  `ConvertStaticToSemaType`. For example, If the import handler fails to load the program (contract), it panics:
https://github.com/onflow/cadence/blob/4ab56fd102820c614d48748dbcfedd9d933344ed/runtime/environment.go#L807-L815

The two tests added cover these two cases; one where the error is returned, and the other for the case where the error is thrown.

Comment (SupunS) [reply runtime/interpreter/interpreter.go]: Extracted the type loading function out, so its no longer a lambda: c236cd4

**Review by dsainati1 — APPROVED** [review]

**Review by turbolent — APPROVED** [review]
Nice!

**Pull request merged**